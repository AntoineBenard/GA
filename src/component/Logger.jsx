import React from 'react';
import PropTypes from 'prop-types';

const Logger = props => (
  <div>
    {props.logs.map(log =>
      (
        <div key={log.line}>
          { props.viewLineNumber &&
            <span>{log.line} -</span>
          }
          {log.value}
        </div>
      ))
    }
  </div>);

Logger.defaultProps = {
  logs: [],
  viewLineNumber: false,
};

Logger.propTypes = {
  logs: PropTypes.arrayOf(PropTypes.object),
  viewLineNumber: PropTypes.bool,
};

export default Logger;
