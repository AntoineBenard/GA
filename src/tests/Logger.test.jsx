import React from 'react';
import renderer from 'react-test-renderer';
import Logger from '../component/Logger';

it('Logger snapshot', () => {
  const tree = renderer
    .create(<Logger logs={[{ line: 1, value: 'coucou' }]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
