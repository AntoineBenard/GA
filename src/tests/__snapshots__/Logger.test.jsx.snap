// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger snapshot 1`] = `
<div>
  <div>
    coucou
  </div>
</div>
`;
